### PR TITLE
Wshorten 64 to 32

### PR DESCRIPTION
**Pull request by akallabeth**
Mostly fixes (unchecked) integer narrowing.
Please have a look and point out where the checks can be replaced by an `assert`

### CODE REVIEW

**Comment by akallabeth** [comment]
@hardening @mfleisz @pnowack last call for reviews, will merge this next week otherwise. (rebasing this gets annoying :grin: )

File: channels/printer/client/printer_main.c [hunk 1]
```diff
@@ -313,7 +312,12 @@ static BOOL printer_load_from_config(const rdpSettings* settings, rdpPrinter* pr
 
 	wlen++;
 	path = get_printer_config_path(settings, wname, wlen * sizeof(WCHAR));
-	PrinterNameLen = wlen * sizeof(WCHAR);
+	{
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply channels/printer/client/printer_main.c:315]: scopes `plen`

File: channels/remdesk/client/remdesk_main.c [hunk 2]
```diff
@@ -185,8 +184,12 @@ static UINT remdesk_write_channel_header(wStream* s, REMDESK_CHANNEL_HEADER* hea
 		ChannelNameW[index] = (WCHAR)header->ChannelName[index];
 	}
 
-	ChannelNameLen = (strnlen(header->ChannelName, sizeof(header->ChannelName)) + 1) * 2;
-	Stream_Write_UINT32(s, ChannelNameLen);        /* ChannelNameLen (4 bytes) */
+	const size_t ChannelNameLen =
+	    (strnlen(header->ChannelName, sizeof(header->ChannelName)) + 1) * 2;
+	if (ChannelNameLen > UINT32_MAX)
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply channels/remdesk/client/remdesk_main.c:189]: good point, asserting the expected result

File: client/SDL/SDL3/sdl_clip.cpp [hunk 3]
```diff
@@ -784,20 +786,20 @@ void sdlClip::ClipCleanCb(void* userdata)
...
-	if (strncmp(mime_mate_copied_files, mime.c_str(), sizeof(mime_mate_copied_files)) == 0)
+	if (strncmp(s_mime_mate_copied_files, mime.c_str(), sizeof(s_mime_mate_copied_files)) == 0)
 		return true;
 	return false;
 }
 
 bool sdlClip::mime_is_text(const std::string& mime)
 {
-	for (size_t x = 0; x < ARRAYSIZE(mime_text); x++)
+	for (size_t x = 0; x < s_mime_text.size(); x++)
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply client/SDL/SDL3/sdl_clip.cpp:800]: done

File: client/common/cmdline.c [hunk 4]
```diff
@@ -2484,7 +2485,7 @@ static int parse_host_options(rdpSettings* settings, const COMMAND_LINE_ARGUMENT
 				return COMMAND_LINE_ERROR_UNEXPECTED_VALUE;
 
 			length = (size_t)(p - arg->Value);
-			if (!freerdp_settings_set_uint32(settings, FreeRDP_ServerPort, lval))
+			if (!freerdp_settings_set_uint32(settings, FreeRDP_ServerPort, (UINT16)lval))
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply client/common/cmdline.c:2488]: done

File: client/SDL/SDL3/sdl_clip.cpp [hunk 5]
```diff
@@ -806,9 +808,9 @@ bool sdlClip::mime_is_text(const std::string& mime)
 
 bool sdlClip::mime_is_image(const std::string& mime)
 {
-	for (size_t x = 0; x < ARRAYSIZE(mime_image); x++)
+	for (size_t x = 0; x < s_mime_image.size(); x++)
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply client/SDL/SDL3/sdl_clip.cpp:811]: done

**Comment by akallabeth** [comment]
> * `WTSVirtualChannelWrite`

ok, fixed all that.
agree with you on the types of internal functions (looks like I´ve overlooked that the first function takes in `int` and modified to the `size_t` used later on)

File: channels/rdpsnd/server/rdpsnd_main.c [hunk 6]
```diff
@@ -82,13 +82,21 @@ static UINT rdpsnd_server_send_formats(RdpsndServerContext* context)
 	}
 
 	const size_t pos = Stream_GetPosition(s);
+	if ((pos < 4) || (pos > UINT16_MAX))
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply channels/rdpsnd/server/rdpsnd_main.c:85]: below the `pos - 4`, but right, should be an assert.

File: channels/rdpei/client/rdpei_main.c [hunk 7]
```diff
@@ -295,14 +299,19 @@ static UINT rdpei_write_pen_frame(wStream* s, const RDPINPUT_PEN_FRAME* frame)
 	return CHANNEL_RC_OK;
 }
 
-static UINT rdpei_send_pen_event_pdu(GENERIC_CHANNEL_CALLBACK* callback, UINT32 frameOffset,
-                                     const RDPINPUT_PEN_FRAME* frames, UINT16 count)
+static UINT rdpei_send_pen_event_pdu(GENERIC_CHANNEL_CALLBACK* callback, size_t frameOffset,
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply channels/rdpei/client/rdpei_main.c:302]: I´d keep it this way, we´ve not finished implementing all the details of this channel (pen redirection is still not really tested)

**Review by mfleisz — APPROVED** [review]

**Comment by VVD** [comment]
After this commit build with Heimdal Kerberos + `WITH_KRB5` fails:
```
/wrkdirs/usr/ports/net/freerdp3/work/freerdp-3.10.0/winpr/libwinpr/sspi/Kerberos/kerberos.c:212:12: error: use of undeclared identifier 'KRB5_ERROR'
  212 |                                 return KRB5_ERROR;
      |                                        ^
```
Build with MIT Kerberos without errors.

P.S. Sorry for doubles.

**Pull request merged**